src/auth/schemas/user.schema.ts

```diff
@@ -0,0 +1,18 @@
+import { Prop, Schema, SchemaFactory } from "@nestjs/mongoose";
+
+
+@Schema({
+    timestamps: true
+})
+export class User {
+    @Prop()
+    name: string
+
+    @Prop({ unique: [ true, 'Duplicate username entered']})
```

Comment:
Can it really be used? i not sure this is can be used. can you test again please. i think you can use `{ unique: true, message: "User duplication error" }` if it not work.